lte/gateway/c/oai/lib/s8_proxy/s8_client_api.cpp

```diff
@@ -0,0 +1,198 @@
+    magma::feg::CreateSessionRequestPgw* csr) {
+  OAILOG_FUNC_IN(LOG_SGW_S8);
+  switch (msg->pdn_type) {
+    case IPv4:
+      csr->set_pdn_type(magma::feg::PDNType::IPV4);
+      csr->mutable_paa()->set_ipv4_address(
+          (char*) &msg->paa.ipv4_address, sizeof(msg->paa.ipv4_address));
+      break;
+    case IPv6:
+      csr->set_pdn_type(magma::feg::PDNType::IPV6);
+      csr->mutable_paa()->set_ipv6_address(
+          (char*) &msg->paa.ipv6_address, sizeof(msg->paa.ipv6_address));
+      csr->mutable_paa()->set_ipv6_prefix(msg->paa.ipv6_prefix_length);
+      break;
+    case IPv4_AND_v6:
+      csr->set_pdn_type(magma::feg::PDNType::IPV4V6);
+      csr->mutable_paa()->set_ipv4_address(
+          (char*) &msg->paa.ipv4_address, sizeof(msg->paa.ipv4_address));
+      csr->mutable_paa()->set_ipv6_address(
+          (char*) &msg->paa.ipv6_address, sizeof(msg->paa.ipv6_address));
+      csr->mutable_paa()->set_ipv6_prefix(msg->paa.ipv6_prefix_length);
+      break;
+    default:
+      std::cout << "[ERROR] Invalid pdn type " << std::endl;
+      break;
+  }
+  OAILOG_FUNC_OUT(LOG_SGW_S8);
+}
+
+static void convert_indication_flag_to_proto_msg(
+    const itti_s11_create_session_request_t* msg,
+    magma::feg::CreateSessionRequestPgw* csr) {
+  OAILOG_FUNC_IN(LOG_SGW_S8);
+#define INDICATION_FLAG_SIZE 3
+  char indication_flag[INDICATION_FLAG_SIZE];
+  indication_flag[0] = (msg->indication_flags.daf << DAF_FLAG_BIT_POS) |
+                       (msg->indication_flags.dtf << DTF_FLAG_BIT_POS) |
+                       (msg->indication_flags.hi << HI_FLAG_BIT_POS) |
+                       (msg->indication_flags.dfi << DFI_FLAG_BIT_POS) |
+                       (msg->indication_flags.oi << OI_FLAG_BIT_POS) |
+                       (msg->indication_flags.isrsi << ISRSI_FLAG_BIT_POS) |
+                       (msg->indication_flags.israi << ISRAI_FLAG_BIT_POS) |
+                       (msg->indication_flags.sgwci << SGWCI_FLAG_BIT_POS);
+
+  indication_flag[1] = (msg->indication_flags.sqci << SQSI_FLAG_BIT_POS) |
+                       (msg->indication_flags.uimsi << UIMSI_FLAG_BIT_POS) |
+                       (msg->indication_flags.cfsi << CFSI_FLAG_BIT_POS) |
+                       (msg->indication_flags.crsi << CRSI_FLAG_BIT_POS) |
+                       (msg->indication_flags.p << P_FLAG_BIT_POS) |
+                       (msg->indication_flags.pt << PT_FLAG_BIT_POS) |
+                       (msg->indication_flags.si << SI_FLAG_BIT_POS) |
+                       (msg->indication_flags.msv << MSV_FLAG_BIT_POS);
+
+  indication_flag[2] = (msg->indication_flags.s6af << S6AF_FLAG_BIT_POS) |
+                       (msg->indication_flags.s4af << S4AF_FLAG_BIT_POS) |
+                       (msg->indication_flags.mbmdt << MBMDT_FLAG_BIT_POS) |
+                       (msg->indication_flags.israu << ISRAU_FLAG_BIT_POS) |
+                       (msg->indication_flags.ccrsi << CCRSI_FLAG_BIT_POS);
```

Comment:
Why are these flags just 5 bits? should the remaining 3 bits be set to zero?

Reply:
As per 29274 spec and section 8.12, https://www.etsi.org/deliver/etsi_ts/129200_129299/129274/15.04.00_60/ts_129274v150400p.pdf 
These bits are spare and made to set zero